### DISCUSSION TITLE
Kill Tracker Keybind

### DISCUSSION THREAD

**Discussion by Eigengrau16161D** [Lailloken/Exile-UI#242] [Ideas & Feedback]
The kill tracker not being automated makes sense, but clicking there immediately is tricky sometimes especially if the portal ends up "hot". Two possible methods I thought of to improve ease of use, depending on limitations-if Lailloken already reads the client.txt (though I don't think it does), if the kill tracker is set to log any instance of the /kills result, then people can use any of their own macros to trigger the /kills command and improve ease of use there. The other option would be to add an omni key function to trigger and log the /kills command. Tapping omni key might be enough assuming there's no other relevant binding to that when no menus are open?

**Comment by Lailloken** [comment]
> but clicking there immediately is tricky sometimes especially if the portal ends up "hot"

What do you mean by that? You generally only have to click it once at the start of the map, then again in the hideout if you know you're done with the map. But yes, your solutions do sound better. Off the top of my head, there shouldn't be an issue with starting/refreshing the tracker by omni-clicking once, but I will do some testing tomorrow.

Only thing I can think of right now is that the first press of the key in a map would be "swallowed," which could be confusing if there was a movement skill bound to it. Same in the hideout. I could theoretically just add a tooltip that follows the cursor as a reminder to press it once to start the tracker, just to avoid confusion (only at the start of the map, not for the hideout -- that could end up being obnoxious).

**Reply by Eigengrau16161D** [reply]
> but clicking there immediately is tricky sometimes especially if the portal ends up "hot"
>
It's more of a minor qol gripe that's been exacerbated by losing accuracy switching to a trackball recently to see if it'll relieve wrist strain from this lovely game XD. My description of "hot" was supposed to refer to when you enter a map/instance that has mobs already ready to go at the entrance. There's load protection, but that drops the second you start moving, which could be caused by missing the timer and clicking to move. Setting up some kind of keybind gives it a tactile/responsive weight, at least to my mind. 

> Only thing I can think of right now is that the first press of the key in a map would be "swallowed,"
>
That's a good consideration. I wonder what the demographics are for where people are binding the omni-key. I was trying to think if there any features of the omni key that were used during active mapping. Leveling tracker is reasonably non-competitive. Betrayal has the screen check. The only other thing I can think of would be if someone for some reason wanted to check out an item mid-map.

**Reply by Lailloken** [reply]
> to see if it'll relieve wrist strain from this lovely game XD

A vertical mouse helped me with that, but your mileage may vary. Bought the cheapest one I could find (8 Euros) just to test if it helped, and it's still holding up after 8 months, which surprised me quite a bit. It had horrible mouse-glides, though, but fortunately I had some spare generic ones. Now, I only use my proper gaming mouse for first/third-person games (because the cheap sensor makes camera movement quite jerky/stuttery).

> The only other thing I can think of would be if someone for some reason wanted to check out an item mid-map.

I can make it specific enough to only activate the "/kills" command when it's actually required, i.e. when entering a map for the very first time. Every subsequent key-press would be unaffected.

**Comment by Eigengrau16161D** [comment]
Hard agree on the vertis, I've been using a Zelotes c-18, but I wanted to find an ergonomic option with 6+ buttons that didn't have a scroll wheel that died in two months. The trackball's been a nice experiment for anything not camera-based, but the polling rate of 125hz means you get those jerky/stuttery movements in fps.

Not sure I can offer more feedback on the kills command, it's more your wheelhouse/time. Thanks for considering it.

**Reply by Lailloken** [reply]
> Not sure I can offer more feedback on the kills command, it's more your wheelhouse/time. Thanks for considering it.

Played around with it a bit. When entering a map for the first time, there will now be a tooltip attached to the cursor that reminds you to press the omni-key to start the kill-tracker. I find it better than clicking the timer and having the panel flash. Getting the final kill-count hasn't changed though, as I don't think this omni-key prompt can be used when leaving a map (it would be annoying if you merely left the map for trading or something).
If you want to test it, I have put up a new dev-build that includes the changes: https://github.com/Lailloken/Lailloken-UI/archive/refs/heads/dev.zip

**Comment by Eigengrau16161D** [comment]
Yeah, this definitely feels better!
